llvm/test/Transforms/CorrelatedValuePropagation/icmp.ll

```diff
@@ -1246,3 +1246,118 @@ declare <2 x i8> @llvm.umin.v2i8(<2 x i8>, <2 x i8>)
 declare <2 x i8> @llvm.umax.v2i8(<2 x i8>, <2 x i8>)
 
 attributes #4 = { noreturn }
+
+define i64 @pr69928(i64 noundef %arg, i64 noundef %arg1) {
+; CHECK-LABEL: @pr69928(
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    [[CMP1:%.*]] = icmp ult i64 [[ARG:%.*]], 64424509440
+; CHECK-NEXT:    [[AND:%.*]] = and i64 [[ARG1:%.*]], 4294967295
+; CHECK-NEXT:    [[CMP2:%.*]] = icmp ult i64 [[ARG]], [[AND]]
+; CHECK-NEXT:    [[SELECT:%.*]] = select i1 [[CMP1]], i1 [[CMP2]], i1 false
+; CHECK-NEXT:    br i1 [[SELECT]], label [[GOOD:%.*]], label [[BAD:%.*]]
+; CHECK:       bad:
+; CHECK-NEXT:    ret i64 1
+; CHECK:       good:
+; CHECK-NEXT:    ret i64 0
+;
+entry:
+  %cmp1 = icmp ult i64 %arg, 64424509440
+  %and = and i64 %arg1, 4294967295
+  %cmp2 = icmp slt i64 %arg, %and
+  %select = select i1 %cmp1, i1 %cmp2, i1 false
+  br i1 %select, label %good, label %bad
+
+bad:
+  ret i64 1
+
+good:
+  ret i64 0
+}
+
+define i1 @test_select_flip(i64 noundef %arg) {
```

Comment:
Drop the noundef arguments, unless they are necessary.

Reply:
`getConstantRangeAtUse` requires noundef/no poison. I don't get why, so keep this flag.

Reply:
Ah yes, you are right. Some at-use transforms may be unsafe with undef values.